test/extensions/filters/http/wasm/test_data/test_grpc_call_cpp.cc

```diff
@@ -72,11 +77,42 @@ FilterHeadersStatus GrpcCallContext::onRequestHeaders(uint32_t, bool end_of_stre
   }
   root()->grpcCallHandler(grpc_service_string, "service", "method", initial_metadata, value, 1000,
                           std::unique_ptr<GrpcCallHandlerBase>(root()->handler_));
+  return FilterHeadersStatus::StopIteration;
+}
+
+class GrpcCallContext : public Context {
+public:
+  explicit GrpcCallContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t, bool) override;
+
+  GrpcCallRootContext* root() { return static_cast<GrpcCallRootContext*>(Context::root()); }
+};
+
+static RegisterContextFactory register_GrpcCallContext(CONTEXT_FACTORY(GrpcCallContext),
+                                                       ROOT_FACTORY(GrpcCallRootContext),
+                                                       "grpc_call");
+
+FilterHeadersStatus GrpcCallContext::onRequestHeaders(uint32_t, bool end_of_stream) {
+  google::protobuf::Value value;
+  value.set_string_value("request");
+  HeaderStringPairs initial_metadata;
+  root()->handler_ = new MyGrpcCallHandler();
   if (root()->grpcCallHandler(
           "bogus grpc_service", "service", "method", initial_metadata, value, 1000,
-          std::unique_ptr<GrpcCallHandlerBase>(new MyGrpcCallHandler())) == WasmResult::Ok) {
-    logError("bogus grpc_service accepted error");
+          std::unique_ptr<GrpcCallHandlerBase>(new MyGrpcCallHandler())) == WasmResult::ParseFailure) {
+    logError("bogus grpc_service rejected");
   }
+  if (end_of_stream) {
+    if (root()->grpcCallHandler("cluster", "service", "method", initial_metadata, value,
+                                1000, std::unique_ptr<GrpcCallHandlerBase>(root()->handler_)) ==
+        WasmResult::Ok) {
+      logError("expected failure did not occur");
+    }
```

Comment:
Please convert this to a positive test and log that failure occured.

Reply:
done